scripts/sidebarJsonToYaml.js

```diff
@@ -0,0 +1,35 @@
+const fs = require('fs')
+const { indent } = require('../utils/indentString')
+const sidebarJSON = require('../content/_data/sidebar.json')
+const keys = Object.keys(sidebarJSON)
+let yamlArray = []
```

Comment:
Please use a YAML parser to convert your data structure to YAML. Don't stringify manually.

https://www.npmjs.com/package/yaml

Reply:
@cowboy I tried use both js-yaml and yaml but none of them are able to output in the specific format we need. https://github.com/cypress-io/cypress-services/blob/develop/packages/on/data/manifest.yml

Which is why I did this manually.

Reply:
The code that is reading `manifest.yml` is just `yaml.load()` from `js-yaml`, so anything that outputs valid YAML should work, including one of the above libraries.

https://github.com/cypress-io/cypress-services/blob/1386378e4d05e38268bc8708a802b4e53af9aeb9/packages/on/lib/manifest.js#L5

Reply:
@cowboy I don't think that is accurate as if you look at the manifest.yml I linked above there needs to be `.html` extension added to each line. I think the formatting is too specific to use a 3rd party library. Believe me I wish I could, as it would make things much easier, but the format they are expecting is too unique.

Reply:
When you create the data structure that will be serialized into YAML, add the `".html"` suffix to the slug, and then serialize the object.

For example, this code:

```js
const input = [
  { slug: 'a' },
  { slug: 'b' },
  { slug: 'c', children: [ { slug: 'd' }, { slug: 'e', children: [ { slug: 'f' } ] } ] },
  { slug: 'g' }
]

const process = arr => arr.reduce((acc, { slug, children }) => ({
  ...acc,
  [slug]: children ? process(children) : `${slug}.html`
}), {})

const output = process(input)

fs.writeFileSync('data/manifest.yml', yaml.dump({ output }), 'utf8')
```

Will generate a `data/manifest.yml` file that looks like:

```yaml
output:
  a: a.html
  b: b.html
  c:
    d: d.html
    e:
      f: f.html
  g: g.html
```

Reply:
@cowboy installed `js-yaml` and used it for the final output, i.e., `yaml.dump()`

Reply:
@robertguss this code is still serializing the YAML data manually. Not only that, but it's now outputting a YAML string, and not a YAML object, like before, so it's likely not working, since the [`normalizeNestedPaths()`](https://github.com/cypress-io/cypress-services/blob/develop/packages/on/lib/manifest.js#L7) function expects an object, and not a string.

We shouldn't ever be writing code that constructs YAML-looking (or JSON-looking) output via string concatenation. It's an anti-pattern, is generally prone to errors, and in most cases, results in significantly more code than is necessary.

Please rewrite this script to generate the proper data structure, which is an object with sub-objects, and then use `yaml.dump()` to serialize that data structure to YAML. Ideally, you'll do this recursively, so that the generated `data/manifest.yml` file will be correct, even if another level of nesting is added (which is actually very possibly going to happen).

Additionally, please don't hard-code in checks against the `title` being `Table of Contents` or `All Assertions`. In those cases, you need to ask yourself if there is something more general you can be checking, to help generate the correct output. However, stepping back for a moment, It's not clear what the correct output should even be in those cases. Have you looked at what's being done with the data in the generated file?